src/commands/flow/spec/flow.spec.ts

```diff
@@ -0,0 +1,187 @@
+          predecessors: []
+        },
+        flow2: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow1']
+        }
+      }
+
+      setupMocks(flows)
+
+      await execute(
+        '',
+        '',
+        '',
+        undefined as any as Target,
+        undefined as any as boolean,
+        undefined as any as SASjs
+      )
+
+      const [flow1, flow2] = Object.values(flows)
+
+      validateFlowExecution([flow1, flow2])
+    })
+
+    it('should execute flow in order - dependent on first', async () => {
+      const flows = {
+        flow1: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow2']
+        },
+        flow2: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: []
+        }
+      }
+
+      setupMocks(flows)
+
+      await execute(
+        '',
+        '',
+        '',
+        undefined as any as Target,
+        undefined as any as boolean,
+        undefined as any as SASjs
+      )
+
+      const [flow1, flow2] = Object.values(flows)
+
+      validateFlowExecution([flow2, flow1])
+    })
+
+    it('should execute flow in order - two level dependency', async () => {
+      const flows = {
+        flow1: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow3']
+        },
+        flow2: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: []
+        },
+        flow3: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow2']
+        }
+      }
+
+      setupMocks(flows)
+
+      await execute(
+        '',
+        '',
+        '',
+        undefined as any as Target,
+        undefined as any as boolean,
+        undefined as any as SASjs
+      )
+
+      const [flow1, flow2, flow3] = Object.values(flows)
+
+      validateFlowExecution([flow2, flow3, flow1])
+    })
+
+    it('should execute flow in order - complex dependency', async () => {
+      const flows = {
+        flow1: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow4']
+        },
+        flow2: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow5', 'flow4']
+        },
+        flow3: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: []
+        },
+        flow4: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow3']
+        },
+        flow5: {
+          jobs: [{ location: 'somejob' }],
+          predecessors: ['flow3']
+        }
+      }
+      setupMocks(flows)
+
+      await execute(
+        '',
+        '',
+        '',
+        undefined as any as Target,
+        undefined as any as boolean,
+        undefined as any as SASjs
+      )
+
+      const [flow1, flow2, flow3, flow4, flow5] = Object.values(flows)
+
+      validateFlowExecution([flow3, flow4, flow1, flow5, flow2])
+    })
+  })
+
+  // it(`should execute 2 chained flows with a failing job in predecessor's flow`, async () => {})
```

Comment:
Are you planning to add these test cases?

Reply:
I thougth about these, only ones left from server.specs, but these are already being covered by modules specs

Reply:
then we need to remove these comments